subsys/bluetooth/controller/ll_sw/ull.c

```diff
@@ -1226,6 +1226,47 @@ void *ull_disable_mark_get(void)
 	return mark_get(mark_disable);
 }
 
+/**
+ * @brief Stops a specified ticker using the ull_disable_(un)mark functions.
+ *
+ * @param ticker_handle The handle of the ticker.
+ * @param param         The object to mark.
+ * @param lll_disable   Optional object when calling @ref ull_disable
+ *
+ * @return 0 if success, else error.
+ */
+uint32_t ull_ticker_stop_with_mark(uint8_t ticker_handle, void *param,
+				   void *lll_disable)
+{
+	uint32_t volatile ret_cb;
+	uint32_t ret;
+	void *mark;
+	mark = ull_disable_mark(param);
+	LL_ASSERT(mark == param);
+
+	ret_cb = TICKER_STATUS_BUSY;
+	ret = ticker_stop(TICKER_INSTANCE_ID_CTLR, TICKER_USER_ID_THREAD,
+			  ticker_handle, ull_ticker_status_give,
+			  (void *)&ret_cb);
+	ret = ull_ticker_status_take(ret, &ret_cb);
+	if (ret) {
+		mark = ull_disable_unmark(param);
+		LL_ASSERT(mark == param);
+
+		return ret;
+	}
+
+	if (lll_disable) {
+		ret = ull_disable(lll_disable);
+		LL_ASSERT(!ret);
+	}
```

Comment:
```suggestion
	ret = ull_disable(lll_disable);
	LL_ASSERT(!ret);
```

Reply:
In some of the cases `lll_disable` is set to NULL (as nothing was disabled in them before) and `lll_disable` expects a non-null argument, so we must guard the call.